docs/src/advanced/generic_types.md

```diff
@@ -30,6 +30,17 @@ purely shorthand for the sake of ergonomics.
 
 ## Trait Constraints
```

Comment:
We should a note in italic at the end of this section that says that
"`where` clauses are still work-in-progress".